src/main/java/appeng/me/cluster/implementations/CraftingCPUCluster.java

```diff
@@ -1242,6 +1265,34 @@ public void breakCluster()
 		}
 	}
 
+	private void prepareETA()
+	{
+		this.lastTime = System.nanoTime();
+		this.elapsedTime = 0;
+		int itemCount = 0;
+
+		IItemList<IAEItemStack> list = AEApi.instance().storage().createItemList();
+		this.getListOfItem( list, CraftingItemList.ACTIVE );
+		this.getListOfItem( list, CraftingItemList.PENDING );
+
+		for( IAEItemStack ge : list )
+		{
+			itemCount += ge.getStackSize();
+		}
+
+		this.startItemCount = itemCount;
+		this.remainingItemCount = itemCount;
+	}
+
+	private void updateETA( IAEItemStack is )
+	{
+		final long nextStartTime = System.nanoTime();
+		this.elapsedTime += nextStartTime - lastTime;
+		this.lastTime = nextStartTime;
+		this.remainingItemCount -= is.getStackSize();
+		System.out.println( this.remainingItemCount + " " + this.elapsedTime );
```

Comment:
Yeah, it is a nightmare to actually find the correct itemstacks and let `remainingItemCount` tend towards 0 and not -5134 or +1234. I am not even convinced that it currently works perfect.